### PR TITLE
First draft of test roadmap plan

### PR DESCRIPTION
**Pull request by JamesWTruher**
I'm looking for comments and suggestions

> ready for merge<

### CODE REVIEW

File: docs/testing-guidelines/TestRoadmap.md [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+* Downlevel Full PowerShell Client -> PowerShell Core Server 
+
+### Challenges
+* On Windows, we have cmdlets to enable and disable remote access, those cmdlets do not exist on non-Windows platforms (they rely on configuration stored in the registry), nor do we support configuration for the SSH protocol.
+We need to be sure that we can easily enable remoting for the non-Windows platforms and support both WSMan _and_ SSH configurations.
+* Our current multi-machine tests do not test the connection code, they simply execute test code remotely and retrieve results and assume a good connection.
+The infrastructure used for these tests is STEX which is not an open environment.
+We will need to create automation to create and configure the test systems in the test matrix and then invoke tests on them.
+It is not clear that our current CI systems can accomodate our needs here as neither AppVeyor or Travis can supply us with all of the OS images needed.
+We may need to create our own heterogeneous environment in Azure, or look to other teams (MS Build Lab) for assistance.
```
Comment (adityapatwardhan) [inline docs/testing-guidelines/TestRoadmap.md]: We should talk to Bryce and find out about the Jenkins system, it may be able to support our needs.

Comment (brycem) [inline docs/testing-guidelines/TestRoadmap.md]: Abolutely - One nice thing about most of your PowerShell WTT test content that I've noticed is that your test development pattern seems to enforce have a clean(ish) separation of your test code from the test infrastructure.  I've been playing with DSC resources to create single and multi-machine configs ([based on the TLG](http://social.technet.microsoft.com/wiki/contents/articles/1262.test-lab-guides.aspx#Base_Configuration_TLG)) in Azure that are optimized to enable faster and (cheaper) clean\resets between runs.  We should sit down and figure out which STEX templates you're most interested in getting running.

File: docs/testing-guidelines/TestRoadmap.md [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+# Reporting
+Currently, we report against the simplest of KPI:
+* is the CI build error free (which is part of the PR/Merge process - and reported on our landing page)
+
+We are also collecting data for a daily build, but not yet report on the following KPI
+* is the daily build error free (we are running this on AppVeyor, we still need to do this for Travis-CI)
+
+There are a number of KPIs which we could report on:
+* Code KPIs
+    * What is the coverage of our `CI` tests
```
Comment (adityapatwardhan) [inline docs/testing-guidelines/TestRoadmap.md]: >coverage [](start = 18, length = 8)

By coverage, do you mean code coverage blocks/arcs or % of cmdlets / parameters sets or some other metric?

File: docs/testing-guidelines/TestRoadmap.md [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+* Create PowerBI queries to visualize our KPIs
+
+# Release Criteria
+We must start defining the release criteria for a production ready release of PowerShell Core, as an initial proposal:
+* No open issues for the release
+* 80% code coverage of high use cmdlets (cmdlets used by 70% of users, as captured via telemetry)
+* 90% code coverage of language elements (coverage error code paths may not be 100%) 
+* 60% Code Coverage on Windows via Github tests
+* Acceptance by 50% PowerShell MVPs (via Survey)
+* Acceptance by Partners (via Survey)
```
Comment (adityapatwardhan) [inline docs/testing-guidelines/TestRoadmap.md]: Should we have some metric for cross-remoting scenario?

File: docs/testing-guidelines/TestRoadmap.md (hunk 2 above)
Comment (JamesWTruher) [inline docs/testing-guidelines/TestRoadmap.md]: i meant % blocks covered - I've updated the language to clarify

**Review by SteveL-MSFT — APPROVED** [review]

**Pull request merged**